# Description: Do not count system inodes into quota for ext[234]
# Author: Jan Kara

--- quota/quotacheck.c	2010-02-18 09:44:10.000000000 +0100
+++ quota-tools//quotacheck.c	2010-05-22 16:47:53.000000000 +0200
@@ -455,8 +455,10 @@
 		return -1;
 	}
 
-	while ((long)i_num) {
-		if (inode.i_links_count) {
+	while (i_num) {
+		if ((i_num == EXT2_ROOT_INO ||
+		     i_num >= EXT2_FIRST_INO(fs->super)) &&
+		    inode.i_links_count) {
 			debug(FL_DEBUG, _("Found i_num %ld, blocks %ld\n"), (long)i_num, (long)inode.i_blocks);
 			if (flags & FL_VERBOSE)
 				blit(NULL);
